File: test/06.test.js
const fs = require('fs');

const { A06, B06 } = require('../lib/06');
const input = fs.readFileSync('./data/06.txt', 'utf-8');

describe('06', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A06(`abc\n\na\nb\nc\n\nab\nac\n\na\na\na\na\n\nb`)).toBe(11);
		});

		it('Solution', () => {
			expect(A06(input)).toBe(6630);
		});
	});

	describe('B', () => {
		it('Example 1', () => {
			expect(B06(`abc\n\na\nb\nc\n\nab\nac\n\na\na\na\na\n\nb`)).toBe(6);
		});

		it('Solution', () => {
			expect(B06(input)).toBe(3437);
		});
	});
});

File: test/06_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/06.dart';

void main() {
  String contents = new File('./data/06.txt').readAsStringSync();
  
  group('A', () {
    test('Example 1', () {
      expect(A06('abc\n\na\nb\nc\n\nab\nac\n\na\na\na\na\n\nb'), equals(11));
    });
    test('Solution', () {
      expect(A06(contents), equals(6630));
    });
  });

  group('B', () {
    test('Example 1', () {
      expect(B06('abc\n\na\nb\nc\n\nab\nac\n\na\na\na\na\n\nb'), equals(6));
    });
    test('Solution', () {
      expect(B06(contents), equals(3437));
    });
  });
}
